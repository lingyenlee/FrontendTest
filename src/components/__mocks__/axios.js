module.exports = {
  get: jest.fn(() => {
    return Promise.resolve({
      data: {
        Brastlewark: [
          {
            id: 1,
            name: "John Doe",
            age: 10,
            weight: 20,
            profession: ["A", "B", "C"],
          },
          {
            id: 2,
            name: "Jane Doe",
            age: 20,
            weight: 30,
            profession: ["D", "E", "F"],
          },
        ],
      },
    });
  }),
};
