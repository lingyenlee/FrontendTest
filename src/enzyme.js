import Enzyme, { configure, shallow, mount, render } from "enzyme";
import Adaptor from "enzyme-adapter-react-16";
import React from "react";

configure({ adapter: new Adaptor() });
export { shallow, mount, render };
export default Enzyme;

// Make Enzyme functions available in all test files without importing
global.React = React;
global.shallow = shallow;
global.render = render;
global.mount = mount;
