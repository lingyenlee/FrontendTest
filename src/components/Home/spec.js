import React from "react";
import { shallow, mount } from "enzyme";
import HomePage from "./Homepage";
import { create } from "react-test-renderer";
import axios from "axios";

describe("Homepage", () => {

  //mock API call
  it("should make an API call", () => {
    jest.mock("axios");
    const getSpy = jest.spyOn(axios, "get");
    mount(<HomePage />);
    expect(getSpy).toBeCalled();
  });

  //test for fetchAnimals
  it("Should call fetchAnimals", () => {
    const getSpy = jest.spyOn(HomePage.prototype, "fetchAnimals");
    mount(<HomePage />);
    expect(getSpy).toHaveBeenCalledTimes(1);
  });

  //test for snapshot of Homepage
  it("Should matches snapshot", () => {
    const component = create(<HomePage />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  //calls for component did mount
  it("Should call componentDidMount", () => {
    jest.spyOn(HomePage.prototype, "componentDidMount");
    //use shallow at first but it doesn't work. why?
    mount(<HomePage />);
    expect(HomePage.prototype.componentDidMount.mock.calls.length).toBe(1);
  });
});
