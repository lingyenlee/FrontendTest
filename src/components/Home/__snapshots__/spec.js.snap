// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Should matches snapshot 1`] = `
<div>
  <div>
    Loading image....
  </div>
</div>
`;
